src/main/java/com/hackathonorganizer/hackathonwriteservice/team/service/TagService.java

```diff
@@ -0,0 +1,73 @@
+package com.hackathonorganizer.hackathonwriteservice.team.service;
+
+import com.hackathonorganizer.hackathonwriteservice.team.exception.ResourceAlreadyExistsException;
+import com.hackathonorganizer.hackathonwriteservice.team.exception.ResourceNotFoundException;
+import com.hackathonorganizer.hackathonwriteservice.team.model.Tag;
+import com.hackathonorganizer.hackathonwriteservice.team.model.dto.TagRequest;
+import com.hackathonorganizer.hackathonwriteservice.team.repository.TagRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TagService {
+
+    private final TagRepository tagRepository;
+
+    public Tag create(TagRequest tagRequest) {
+        validateName(tagRequest.name());
+        val tagToSave = Tag.builder()
+                .name(tagRequest.name())
+                .build();
+        return tagRepository.save(tagToSave);
+    }
+
+    public Tag editById(Long id, TagRequest tagRequest) {
+        validateName(tagRequest.name());
+        return tagRepository.findById(id)
+                .map(tagToEdit -> {
+                    tagToEdit.setName(tagRequest.name());
+                    return tagRepository.save(Tag.builder()
```

Comment:
Moze przeniesc to do prywatnej metody?

Reply:
Dlaczego?

Reply:
Ktos mnie kiedys uczyl ze nie powinno sie bezposrednio wywolywac repo itp, tylko przenosci do prywatnych metod, szczegolnie jak kod sie powtarza w innych miejscach, ale to niech ktos jeszcze sie tu wypowie

Reply:
Jaka różnica jak wywolam z repo bezpośrednio save a 5 razy z prywatnej metody? Według mnie żadna a tylko przeszkadza w czytaniu kodu bo muszę patrzeć co ta metoda robi a tam tylko save? Osobiście nie widzę sensu w tym żadnego

Reply:
Nwm, czy to nie mozna podciagnac pod DRY, no ale najlepiej to przedyskutowac na zebraniu

Reply:
już w drugiej CRce na to odpowiedziałem, tu dorzucę tylko że repo nie powinno się eksponować poza dany package, czyli np TagService nie powinien dotykać HackatonRepository. hackaton repo nie tyle może co musi być użyte w serwisie. chyba źle rozumiesz zamysł DRY bo w nim chodzi o duplikację kodu w kilku różnych miejsach, a zamiana jednej linijki na inną linijkę która tą pierwszą tylko owrapuje to nie jest żadne DRY